metricbeat/module/elasticsearch/index_recovery/_meta/data.json

```diff
@@ -1,38 +1,137 @@
 {
-    "@timestamp": "2017-10-12T08:05:34.853Z",
-    "agent": {
-        "hostname": "host.example.com",
-        "name": "host.example.com"
+    "@timestamp": "2020-08-26T20:34:41.034Z",
+    "@metadata": {
+        "beat": "metricbeat",
+        "type": "_doc",
+        "version": "8.0.0"
+    },
+    "service": {
+        "name": "elasticsearch",
+        "address": "localhost:9200",
+        "type": "elasticsearch"
+    },
+    "event": {
+        "dataset": "elasticsearch.index.recovery",
+        "module": "elasticsearch",
+        "duration": 30884428
+    },
+    "metricset": {
+        "name": "index_recovery",
+        "period": 10000
     },
     "elasticsearch": {
         "cluster": {
-            "id": "3LbUkLkURz--FR-YO0wLNA",
-            "name": "es1"
+            "name": "docker-cluster",
+            "id": "giRKn5pZQ0afY0YrRiXdxQ"
         },
         "index": {
-            "name": ".monitoring-es-6-2018.11.20",
             "recovery": {
-                "id": 0,
-                "primary": true,
-                "source": {},
-                "stage": "DONE",
-                "target": {
-                    "host": "127.0.0.1",
-                    "id": "FMRmkE3HTU6xxxoFK-06Ww",
-                    "name": "es1_1"
-                },
-                "type": "EMPTY_STORE"
+                "shards": [
```

Comment:
Hmmm, this is a tricky change.

This appears to be a breaking change too. Not so much defining a new `elasticsearch.index.recovery.shards` field because that's an additive change. But more that we're taking away the `elasticsearch.index.recovery.id`, `elasticsearch.index.recovery.primary`, etc. fields.

I think the reason you had to do this is because previously each document produced by the `index_recovery` metricset represented a single shard, whereas what Stack Monitoring expects each document to contain an array of shards. 

@chrisronline what do you need the data type of this `elasticsearch.index.recovery.shards` field to be for queries to work correctly: `object` or `nested`? Depending on that we might need to figure out how we handle this breaking change.

Reply:
We don't perform any aggregations across this data, so `object` works fine.

Reply:
Okay, that's good news. 

I'm still concerned about the breaking change in document structure here, for any existing users of the `elasticsearch/index_recovery` metricset (non-xpack). 

@chrisronline How much work would it be to make the UI work on the current (before this PR) structure of `elasticsearch/index_recovery` metricset (non-xpack) documents where one document == one shard? Keep in mind that this would be code we'd get to clean up in 8.0 as we could introduce a breaking change in document structure in this metricset at that point.

Reply:
This should be possible. We only access this data from a single code path and adding any kind of funky logic in there is fine, knowing it can go away in 8.0.

Reply:
Thanks so much @chrisronline  ❤️ . Essentially you'll need logic to check for the new structure (1 document == 1 shard) that will be produced by this metricset when its run without the `xpack.enabled` flag but fall back to the current logic of expecting multiple shard inner documents in a single `type: index_recovery` document. Hope that makes sense.

In that case, @sayden, we should keep the 1 document == 1 shard structure that is currently (prior to this PR) being produced by this metricset. We can add more fields to that document, if needed for Stack Monitoring, of course.

Reply:
@ycombinator Yes, that's my understanding. Should be fine

Reply:
Yeah! Ok! Sounds good! I'll revert to the OSS original structure of 1 doc == 1 shard.